File: app/src/main/java/com/coppermobile/mysamplemvvmdatabindinglivedata/viewmodels/CommentsViewModel.java
package com.coppermobile.mysamplemvvmdatabindinglivedata.viewmodels;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import com.coppermobile.mysamplemvvmdatabindinglivedata.data.Comments;
import com.coppermobile.mysamplemvvmdatabindinglivedata.repository.CommentsRepository;

import java.util.List;

public class CommentsViewModel extends AndroidViewModel {

    private final CommentsRepository commentsRepository;

    public CommentsViewModel(@NonNull Application application, CommentsRepository commentsRepository) {
        super(application);
        this.commentsRepository = commentsRepository;
    }

    public LiveData<List<Comments>> getAllComments(int dishId) {
        return commentsRepository.getComments(dishId);
    }

    public void deleteComment(Comments comments){
        commentsRepository.deleteData(comments);
    }

    public void insertComment(Comments comments){
        commentsRepository.insertData(comments);
    }
}

File: app/src/main/java/com/coppermobile/kotlinsamplemvvmlivedataroom/viewmodels/CommentsViewModel.kt
package com.coppermobile.kotlinsamplemvvmlivedataroom.viewmodels

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData
import com.coppermobile.kotlinsamplemvvmlivedataroom.data.Comments
import com.coppermobile.kotlinsamplemvvmlivedataroom.repository.CommentsRepository

class CommentsViewModel(application: Application, private val commentsRepository: CommentsRepository) : AndroidViewModel(application) {

    fun deleteComment(comment: Comments) {
        commentsRepository.deleteData(comment)
    }

    fun insertComment(comment: Comments) {
        commentsRepository.insertData(comment)
    }

    fun getComments(dishId:Int): LiveData<List<Comments>>? {
       return commentsRepository.getComments(dishId)
    }
}